internal/common/resource.go

```diff
@@ -34,7 +39,7 @@ func CollectResourceStatus(request *Request, funcs ...ReconcileFunc) ([]Resource
 }
 
 type ResourceUpdateFunc = func(expected, found client.Object)
-type ResourceStatusFunc = func(resource client.Object) ResourceStatus
+type ResourceStatusFunc = func(resource client.Object) ReconcileResult
```

Comment:
This function should still return `ResourceStatus`. It is used to create resource specific status, for example, on deployment waiting until its pods are ready.

Reply:
yep combined with your previous comment this works great.